### ISSUE TITLE
INTERFACE_LIBRARY targets may only have whitelisted properties.  The   property "COTIRE_UNITY_TARGET_NAME" is not allowed.

### ISSUE THREAD

**Issue by DoDoENT** [sakra/cotire#87]
I am trying out cotire in my project (instead of custom PCH hack which has some problems) and I am getting this error:

```
CMake Error at cmake/cotire.cmake:3085 (get_target_property):
  INTERFACE_LIBRARY targets may only have whitelisted properties.  The
  property "COTIRE_UNITY_TARGET_NAME" is not allowed.
Call Stack (most recent call first):
  cmake/cotire.cmake:3116 (cotire_map_libraries)
  cmake/cotire.cmake:3213 (cotire_target_link_libraries)
```

Any idea how to fix this? I am using CMake v3.4.1 for Mac OS X.

**Comment by jcelerier** [comment]
It sounds like you're trying to apply `cotire` to an interface library, i.e. one that was added like : 

```
add_library(foo INTERFACE)
```
This won't work due to restrictions in CMake I think (a possibility would be to rename COTIRE_UNITY_TARGET_NAME into INTERFACE_COTIRE_blah) but generally it does not really make sense because an interface library is just here to transport some kind of information when `target_link_libraries`'ed to another, not to be actually built.

**Comment by DoDoENT** [comment]
This happens to me when applying `cotire` to either executable or static library which has interface library as a dependency specified with `target_link_libraries`.

The full stack trace is:

```
CMake Error at cmake/cotire.cmake:3085 (get_target_property):
  INTERFACE_LIBRARY targets may only have whitelisted properties.  The
  property "COTIRE_UNITY_TARGET_NAME" is not allowed.
Call Stack (most recent call first):
  cmake/cotire.cmake:3116 (cotire_map_libraries)
  cmake/cotire.cmake:3213 (cotire_target_link_libraries)
  cmake/common_utils.cmake:87 (cotire)
  CoreUtils/CMakeLists.txt:107 (build_static_library)
```

The `build_static_library` macro is implemented as follows:

```
macro(build_static_library libname)
    include_directories(${CMAKE_CURRENT_SOURCE_DIR}/${SOURCE_DIR})
    set(SOURCES ${ARGN})
    add_library(${libname} STATIC ${SOURCES})
    target_include_directories(${libname} PUBLIC ${CMAKE_CURRENT_SOURCE_DIR}/${SOURCE_DIR})
    target_link_libraries(${libname} ${DEPENDS} ${LIBS})
    if(${USE_PCH})
        cotire(${libname})
    endif()
endmacro()
```

`${DEPENDS}` array contains targets current target depends on. It can contain both defined targets or `IMPORTED` targets (most of `IMPORTED` targets are `INTERFACE`, and some are `STATIC`).

I am using Ninja generator with CMake v3.4.1 on MacOSX.

**Comment by GamePad64** [comment]
Well, I have this bug, too. :+1:

**Comment by sakra** [comment]
Fixed in 1.7.7.
